### DISCUSSION TITLE
How to create a "step by step guide" extension for vscode?

### DISCUSSION THREAD

**Discussion by code1704** [microsoft/vscode-discussions#827] [Extension Development QnA] [answered]
like this one, but for vscode as a vscode extension:
<img width="1107" alt="image" src="https://github.com/microsoft/vscode-discussions/assets/110730540/8ffdb115-ab88-4d60-bd4b-18ff7680445a">

**Comment by alefragnani** [comment]
Hi @code1704 ,

You should take a look at the [Walkthroughs API](https://code.visualstudio.com/api/ux-guidelines/walkthroughs) which appears on the Getting Started page and are automatically opened on install of your extension and provide a convenient way to introduce users to features of your extension. The VS Code Extensions Samples repo provides a [sample](https://github.com/microsoft/vscode-extension-samples/tree/main/getting-started-sample), and you can find more details here https://code.visualstudio.com/api/references/contribution-points#contributes.walkthroughs.

This is a simple yet powerful way to provide a consistent experience for onboarding users. I have created walkthroughs for some of my extensions, with images, interactions, and fully localizables. Here are the PRs that I made for my extensions:
* https://github.com/alefragnani/vscode-language-pascal/pull/104
* https://github.com/alefragnani/vscode-bookmarks/pull/621
* https://github.com/alefragnani/vscode-project-manager/pull/692

Hope this helps

**Reply by code1704** [reply]
Thanks.

The walkthrough seems to be too simple and the user experience is not good enough for fresh users. Can we do it like introjs does?

**Reply by alefragnani** [reply, marked as answer]
No, you can't. At least not with steps located/focusing on VS Code UI elements.

You can however, create a [webview](https://code.visualstudio.com/api/extension-guides/webview) and develop a step by step guide _inside of it_. And, to better simulate VS Code elements, you could use the [Webview UI Toolkit for Visual Studio Code](https://code.visualstudio.com/blogs/2021/10/11/webview-ui-toolkit).

Hope this helps

**Reply by code1704** [reply]
thanks
